src/utils.js

```diff
@@ -71,3 +71,6 @@ utils.sleep = function(time=0) {
     return new Promise(resolve => setTimeout(resolve, time));
 };
 
+utils.isNetsBloxDomain = function (url) {
+    return !!url.match(/^(?:\w+:\/+)?[^/]*\bnetsblox\.org\b\/?/i)
```

Comment:
Technically, this could still generate false positives as something like `somesite.com?param=netsblox.org` would be considered a netsblox domain. That said, it's probably best to leave it as-is since I can't think of many cases this would actually be a problem outside of something like a simple proxy service (in which case, we would want to add the header).

Reply:
Ah, forgot about query strings. Fortunately, even if some site used this pattern, it would only be a problem if they also prevent the extra `X-Source` header, and if so we can just use `[^/?]*`.